pf/tests/extract_inputs_test.go

```diff
@@ -0,0 +1,669 @@
+)
+
+func TestExtractInputsFromOutputsPF(t *testing.T) {
+	type testCase struct {
+		name      string
+		props     resource.PropertyMap
+		resSchema rschema.Schema
+		expect    autogold.Value
+	}
+
+	testCases := []testCase{
+		{
+			name:  "string",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": "bar"}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.StringAttribute{Optional: true},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{
+				resource.PropertyKey("__defaults"): resource.PropertyValue{
+					V: []resource.PropertyValue{},
+				},
+				resource.PropertyKey("foo"): resource.PropertyValue{V: "bar"},
+			}),
+		},
+		// TODO[pulumi/pulumi-terraform-bridge#2218]: This should not yield values for foo in the inputs.
+		{
+			name:  "string with default",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": "bar"}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.StringAttribute{Optional: true, Default: stringdefault.StaticString("bar")},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{
+				resource.PropertyKey("__defaults"): resource.PropertyValue{
+					V: []resource.PropertyValue{},
+				},
+				resource.PropertyKey("foo"): resource.PropertyValue{V: "bar"},
+			}),
+		},
+		{
+			name:  "string with empty value",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": ""}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.StringAttribute{Optional: true},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{resource.PropertyKey("__defaults"): resource.PropertyValue{
+				V: []resource.PropertyValue{},
+			}}),
+		},
+		{
+			name:  "string computed",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": "bar"}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.StringAttribute{Computed: true},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{resource.PropertyKey("__defaults"): resource.PropertyValue{
+				V: []resource.PropertyValue{},
+			}}),
+		},
+		{
+			name:  "list attribute",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": []interface{}{"bar"}}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.ListAttribute{
+						Optional:    true,
+						ElementType: types.StringType,
+					},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{
+				resource.PropertyKey("__defaults"): resource.PropertyValue{
+					V: []resource.PropertyValue{},
+				},
+				resource.PropertyKey("foo"): resource.PropertyValue{V: []resource.PropertyValue{{
+					V: "bar",
+				}}},
+			}),
+		},
+		// TODO[pulumi/pulumi-terraform-bridge#2218]: This should not yield values for properties with defaults in the inputs.
```

Comment:
Does this mean the test is currently testing/expecting undesired state? 

I would prefer seeing a commented-out test that is currently failing, and then the TODO being enabling it once #2218 has been addressed.

Reply:
That's also how I found https://github.com/pulumi/pulumi-terraform-bridge/issues/2180 affects PF